[Debugger] Move resize logout outside of core expandItem/collapseItem…

jstedfast opened the pull request:
… logic

Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1002173/

mrward commented:
I do not see any debugger tooltips with this branch. The IDE log has the following error when I hover over something in the text editor:

```
Async operation failed
System.AggregateException: One or more errors occurred. (Value cannot be null.
Parameter name: value) ---> System.ArgumentNullException: Value cannot be null.
Parameter name: value
  at AppKit.NSControl.set_Font (AppKit.NSFont value) [0x00058] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.16.1.24/src/Xamarin.Mac/AppKit/NSControl.g.cs:1091 
  at MonoDevelop.Debugger.MacDebuggerObjectCellViewBase.UpdateFont (AppKit.NSControl control, System.Int32 sizeDelta) [0x00032] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs:168 
  at MonoDevelop.Debugger.MacDebuggerObjectNameView.UpdateContents () [0x002d0] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectNameView.cs:133 
  at MonoDevelop.Debugger.MacDebuggerObjectCellViewBase.set_BackgroundStyle (AppKit.NSBackgroundStyle value) [0x00007] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs:182 
  at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSendSuper_IntPtr_IntPtr_UInt64(intptr,intptr,intptr,intptr,ulong)
  at AppKit.NSOutlineView.InsertItems (Foundation.NSIndexSet indexes, Foundation.NSObject parent, AppKit.NSTableViewAnimation animationOptions) [0x00099] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.16.1.24/src/Xamarin.Mac/AppKit/NSOutlineView.g.cs:407 
  at MonoDevelop.Debugger.MacObjectValueTreeViewDataSource.Append (MonoDevelop.Debugger.ObjectValueNode node) [0x00099] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeViewDataSource.cs:267 
  at MonoDevelop.Debugger.MacObjectValueTreeView.MonoDevelop.Debugger.IObjectValueTreeView.Appended (MonoDevelop.Debugger.ObjectValueNode node) [0x00000] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeView.cs:593 
  at MonoDevelop.Debugger.ObjectValueTreeViewController+<>c__DisplayClass47_0.<AddValue>b__0 () [0x0001e] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/ObjectValueTreeViewController.cs:245 
  at MonoDevelop.Core.Runtime.RunInMainThread (System.Action action) [0x0000d] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:332 
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.ArgumentNullException: Value cannot be null.
Parameter name: value
  at AppKit.NSControl.set_Font (AppKit.NSFont value) [0x00058] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.16.1.24/src/Xamarin.Mac/AppKit/NSControl.g.cs:1091 
  at MonoDevelop.Debugger.MacDebuggerObjectCellViewBase.UpdateFont (AppKit.NSControl control, System.Int32 sizeDelta) [0x00032] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs:168 
  at MonoDevelop.Debugger.MacDebuggerObjectNameView.UpdateContents () [0x002d0] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectNameView.cs:133 
  at MonoDevelop.Debugger.MacDebuggerObjectCellViewBase.set_BackgroundStyle (AppKit.NSBackgroundStyle value) [0x00007] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs:182 
  at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSendSuper_IntPtr_IntPtr_UInt64(intptr,intptr,intptr,intptr,ulong)
  at AppKit.NSOutlineView.InsertItems (Foundation.NSIndexSet indexes, Foundation.NSObject parent, AppKit.NSTableViewAnimation animationOptions) [0x00099] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.16.1.24/src/Xamarin.Mac/AppKit/NSOutlineView.g.cs:407 
  at MonoDevelop.Debugger.MacObjectValueTreeViewDataSource.Append (MonoDevelop.Debugger.ObjectValueNode node) [0x00099] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeViewDataSource.cs:267 
  at MonoDevelop.Debugger.MacObjectValueTreeView.MonoDevelop.Debugger.IObjectValueTreeView.Appended (MonoDevelop.Debugger.ObjectValueNode node) [0x00000] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeView.cs:593 
  at MonoDevelop.Debugger.ObjectValueTreeViewController+<>c__DisplayClass47_0.<AddValue>b__0 () [0x0001e] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/ObjectValueTreeViewController.cs:245 
  at MonoDevelop.Core.Runtime.RunInMainThread (System.Action action) [0x0000d] in /Users/runner/runners/2.159.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:332 <---

```

jstedfast commented:
Grrrr... yesterday before committing this, I switched back to the master branch and then noticed that I had a bunch of changed files. I thought I reverted all the changes that I didn't want, but it looks like I missed some changes.

It looks like MonoDevelop will do all sorts of broken things with your git repo if you change branches in a Terminal window with the project open.
